src/pages/iv-materials-tfr/Form/MaterialsTransferForm.js

```diff
@@ -0,0 +1,913 @@
+          }
+          invalidate()
+        } else {
+          if (formik.values.notificationGroupId) {
+            notificationData = {
+              recordId: res.recordId,
+              functionId: SystemFunction.MaterialTransfer,
+              notificationGroupId: formik.values.notificationGroupId,
+              date: formik.values.date,
+              reference: formik.values.reference,
+              status: 1
+            }
+
+            await postRequest({
+              extension: AccessControlRepository.Notification.set,
+              record: JSON.stringify(notificationData)
+            })
+          } else if (!formik.values.notificationGroupId) {
+            const data = {
+              recordId: res.recordId,
+              functionId: SystemFunction.MaterialTransfer,
+              notificationGroupId: 0
+            }
+
+            await postRequest({
+              extension: AccessControlRepository.Notification.del,
+              record: JSON.stringify(data)
+            })
+          }
+
+          toast.success(platformLabels.Edited)
+        }
+      } catch (error) {}
+    }
+  })
+
+  const isPosted = formik.values.status === 3
+  const isClosed = formik.values.wip === 2
+  const editMode = !!formik.values.recordId
+
+  const getWeightAndMetalId = async itemId => {
+    const res = await getRequest({
+      extension: InventoryRepository.PP.get,
+      parameters: '_itemId=' + itemId
+    })
+
+    return {
+      weight: res?.record?.weight ?? 0,
+      metalId: res?.record?.metalId
+    }
+  }
+
+  const getUnitCost = async itemId => {
+    const res = await getRequest({
+      extension: InventoryRepository.Cost.get,
+      parameters: '_itemId=' + itemId
+    })
+
+    return res?.record?.currentCost
+  }
+
+  function calcTotalCost(rec) {
+    if (rec.priceType === 1) return rec.qty * rec.unitCost
+    else if (rec.priceType === 2) return rec.qty * rec.unitCost * rec.volume
+    else if (rec.priceType === 3) return rec.qty * rec.unitCost * rec.weight
+    else return 0
+  }
+
+  function calcUnitCost(rec, totalCost) {
+    if (rec.priceType === 1) {
+      rec.unitCost = totalCost / rec.qty
+    } else if (rec.priceType === 2) {
+      rec.unitCost = totalCost / (rec.qty * rec.volume)
+    } else if (rec.priceType === 3) {
+      rec.unitCost = totalCost / (rec.qty * rec.weight)
+    } else {
+      rec.unitCost = 0
+    }
+
+    return rec.unitCost
+  }
+
+  const { totalQty, totalCost, totalWeight } = formik?.values?.transfers?.reduce(
+    (acc, row) => {
+      const qtyValue = parseFloat(row?.qty) || 0
+      const totalCostValue = parseFloat(row?.totalCost) || 0
+      const weightValue = parseFloat(row?.weight) || 0
+
+      return {
+        totalQty: acc.totalQty + qtyValue,
+        totalCost: acc.totalCost + totalCostValue,
+        totalWeight: acc.totalWeight + weightValue
+      }
+    },
+    { totalQty: 0, totalCost: 0, totalWeight: 0 }
+  )
+
+  const getMeasurementUnits = async msId => {
```

Comment:
u can directly
return  await getRequest({
        extension: InventoryRepository.MeasurementUnit.qry,
        parameters: `_msId=${msId}`
      })
      
      and remove try/catch